commands/init.go

```diff
@@ -94,6 +115,113 @@ func resetRoot(root string, saveKeys bool) {
 	}
 }
 
+// RequireInit should be called at the beginning of a command to make
+// sure that the client is initialized.
+//
+//  if err := commands.RequireInit(cmd); err != nil {
+//    return err
+//  }
+//
+// I tried using PersistentPreRun,
+// but it seems that it is called for the commands before root,
+// so viper is not set up when this was called.
+func RequireInit(cmd *cobra.Command) error {
+	root := viper.GetString(cli.HomeFlag)
+	init, err := WasInited(root)
+	if err != nil {
+		return err
+	}
+	if !init {
+		return errors.Errorf("You must run '%s init' first", cmd.Root().Name())
+	}
+	return nil
+}
+
+// WasInited returns true if a light-client was previously initialized
+// in this directory.  Important to ensure proper behavior.
+//
+// Returns error if we have filesystem errors
+func WasInited(root string) (bool, error) {
+	// make sure there is a directory here in any case
+	os.MkdirAll(root, dirPerm)
+
+	// check if there is a config.toml file
+	cfgFile := filepath.Join(root, "config.toml")
+	_, err := os.Stat(cfgFile)
+	if os.IsNotExist(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, errors.WithStack(err)
+	}
+
+	// check if there are non-empty checkpoints and validators dirs
+	dirs := []string{
+		filepath.Join(root, files.CheckDir),
+		filepath.Join(root, files.ValDir),
+	}
+	// if any of these dirs is empty, then we have no data
+	for _, d := range dirs {
+		empty, err := isEmpty(d)
+		if err != nil {
+			return false, err
+		}
+		if empty {
+			return false, nil
+		}
+	}
+
+	// looks like we have everything
+	return true, nil
+}
+
+func checkGenesis(cmd *cobra.Command) error {
+	genesis := viper.GetString(GenesisFlag)
+	if genesis == "" {
+		return nil
+	}
+
+	doc, err := tcmd.ParseGenesisFile(genesis)
```

Comment:
Should this really be in `tendermint/cmd/commands` ? Would it serve better in `tendermint/types` ?